src/coreclr/md/inc/liteweightstgdb.h

```diff
@@ -69,6 +69,21 @@ class CLiteWeightStgdb
 template <class MiniMd>
 void CLiteWeightStgdb<MiniMd>::Uninit()
 {
+    // Clean up allocated column definitions (same logic as CMiniMdBase destructor)
```

Comment:
Why do we need to duplicate this logic here? What prevents CMiniMdBase  destructor from cleaning up the memory?

Reply:
The issue is that `MDInternalRO` and `MDInternalRW` explicitly call `Uninit()` before the automatic `CMiniMdBase` destructor runs. Either that confuses valgrind or it indeed leaves the object orphaned is unclear.

Reply:
Ideally we would be able to rely on the destructor as designed. I actively dislike these `Uninit()` APIs, at best they should be an internal implementation detail.